### PR TITLE
feat: security support added to operation and operationTrait 

### PR DESCRIPTION
**Pull request by LokeshRishi**
2 passing (789ms)

----------|---------|----------|---------|---------|-------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
----------|---------|----------|---------|---------|-------------------
All files |     100 |      100 |     100 |     100 |                   
 index.js |     100 |      100 |     100 |     100 |                   
----------|---------|----------|---------|---------|-------------------

**Description**
Updates to include 'security' at operation and operationTrait level.

**Related issue(s)**
https://github.com/asyncapi/spec/pull/584

### CODE REVIEW

**Comment by smoya** [comment]
Thanks for the PR @LokeshRishi! would you mind changing the title of this PR adding a prefix to be compliant with [Conventional Commits](https://www.conventionalcommits.org)? `feat:` prefix seems the right one.

File: schemas/2.4.0.json
Comment (smoya) [inline schemas/2.4.0.json:619]: Would it make sense to extract this into a new definition called `SecurityRequirements`? Then you will just need to use it by `"$ref": "#/definitions/SecurityRequirements"`

Comment (sekharbans-ebay) [reply schemas/2.4.0.json:619]: > Would it make sense to extract this into a new definition called `SecurityRequirements` or similar? Then you will just need to use it by `"$ref": "#/definitions/SecurityRequirements"`

Hi @smoya  - Its an array of SecurityRequirements. Not sure if just indicating an array of a complex schema warrants a new definition. It would apply to all similar constructs (tags for example). Not sure it serves much purpose.

Comment (smoya) [reply schemas/2.4.0.json:619]: Fair enough 👍 I agree we would need to be consistent and lot of changes will need to be applied.

**Review by smoya — APPROVED** [review]
LGTM, however we need a +1 of a Code Owner.

**Review by dalelane — APPROVED** [review]
LGTM

**Review by derberg — APPROVED** [review]
@fmvilas only your approval is missing

I put https://github.com/asyncapi/spec/pull/584 on Stage 2, 
once we merge this and parser PR, then we can set Stage 3 and merge

**Pull request merged**
